src/js/src/components/SearchFacetOptions.vue

```diff
@@ -42,7 +62,35 @@ export default {
     ...mapActions('Search', {
       updateSolrSettingOffset:'updateSolrSettingOffset',
       addToSearchAppliedFacets:'addToSearchAppliedFacets',
+      addSpecificRequestedFacets:'addSpecificRequestedFacets'
     }),
+    determineFacetAction(facet, length) {
+      if(length === 20) {
+        this.requestAdditionalFacets(facet)
+      }
+      else if(length > 20) {
+        this.removeAdditionalFacets(facet)
+      }
+    },
+    determineText(facet, length) {
+      return length <= 20 ? 'more ' + facet + 's' : 'less ' + facet + 's'
+    },
+    requestAdditionalFacets(facetArea) {
+      this.extraFacetsLoading = true
+      let structuredQuery = this.query + this.searchAppliedFacets.join('')
+      requestService.getAddonFacets(facetArea, this.query).then(result => (this.constructNewFacets(result, facetArea), this.extraFacetsLoading = false), error => (console.log('No additional facets found.'),this.extraFacetsLoading = false))
+    },
+    removeAdditionalFacets(facetArea) {
+      let structuredQuery = this.query + this.searchAppliedFacets.join('')
+      let facets = JSON.parse(JSON.stringify(this.facets))
+      facets['facet_fields'][facetArea] = facets['facet_fields'][facetArea].splice(0,20)
+      this.addSpecificRequestedFacets(facets)
+    },
+    constructNewFacets(result, facetArea) {
+      let facets = JSON.parse(JSON.stringify(this.facets))
+      facets['facet_fields'][facetArea] = result.facet_counts.facet_fields[facetArea]
+      this.addSpecificRequestedFacets(facets)
+    },
```

Comment:
This is, unfortunately, a very good point. It could - and probably should... At least the part where the new facets are added. The removal is, imho, fine called from the template, but the logic for adding the new ones could be mixed in with getting the facets in the store instead. Good point. Damnit. I'll get that fixed.